servant-server/src/Servant/Server.hs

```diff
@@ -126,13 +145,20 @@ import           Servant.Server.Internal
 -- > main :: IO ()
 -- > main = Network.Wai.Handler.Warp.run 8080 app
 --
-serve :: (HasServer api '[]) => Proxy api -> Server api -> Application
+serve :: (HasServer api DefaultErrorFormatters) => Proxy api -> Server api -> Application
 serve p = serveWithContext p EmptyContext
 
-serveWithContext :: (HasServer api context)
+-- | Like 'serve', but allows you to pass custotm context.
+--
+-- 'defaultErrorFormatters' will always be appended to the end of the passed context,
+-- but if you pass your own formatter, it will override the default one.
+serveWithContext :: (HasServer api (context .++ DefaultErrorFormatters), HasContextEntry (context .++ DefaultErrorFormatters) NotFoundErrorFormatter)
```

Comment:
Ah, I had thought that we could get away with a `HasServer api context` signature here. But I'm now seeing how the blanket is to short, and if we try to cover this, something else is going to get uncovered (in particular, HasServer will need to variations, or all the class methods will have to change).

Reply:
Yes, unfortunately looks like there is no sane way to have `HasServer api context` here...